src/main/kotlin/org/rust/ide/refactoring/inlineFunction/RsInlineFunctionProcessor.kt

```diff
@@ -209,7 +217,50 @@ class RsInlineFunctionProcessor(
         }
     }
 
-    private fun addFunctionBodyToCallerWithCorrectSpacing(leftSibling: PsiElement?, rightSibling: PsiElement?, blockParent: RsBlock, body: RsBlock, childContainingEnclosingStatement: PsiElement?) {
+    private fun replaceParameterUsages(functionDup: RsFunction, caller: PsiElement, funcScope: LocalSearchScope) {
+        val parameterBindingsList = functionDup.valueParameters
+            .filter { it !is RsSelfParameter }
+            .map { it.pat }
+            .filterIsInstance(RsPat::class.java)
+
+        val valueArgumentList = when (caller) {
+            is RsCallExpr -> caller.valueArgumentList
+            is RsDotExpr -> caller.methodCall?.valueArgumentList ?: return
+            else -> return
+        }
+        for ((index, parameter) in parameterBindingsList.withIndex()) {
+            var argument = valueArgumentList.exprList[index]
```

Comment:
If there is a function call without enough arguments (e.g. `fn foo(a: u32)` and you call with `foo()`), the refactoring shows a warning (conflict dialog). The user can proceed with the warning however and in that case this code would crash. Assume that there may not be enough arguments and do not use `[]` indexing here.

Reply:
Didn't see a safe way to return (`[]`, `get()`, and `elementAt()` all throw exceptions), so I created an *if* statement that returns if `index` surpasses the list's size.

Reply:
You can use [this](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/get-or-null.html) :)

Reply:
Yeah, I know about this and `getOrElse()`, they're just delaying the problem a single line, to then *return* when we see a *null*.
I prefer to *return* just as I see that  there aren't enough arguments.

Reply:
You could use something like `var argument = valueArgumentList.exprList.getOrNull(index) ?: return` or just return before the cycle, since the sizes are known beforehand, but that's a detail :)